docs/options api.md

```diff
@@ -0,0 +1,184 @@
+
+Archipelago will be abbreviated as "AP" from now on.
+
+## Option Definitions
+Option parsing in AP is done using different Option classes. For each option you would like to have in your game, you
+need to create:
+- A new option class with a docstring detailing what the option will do to your user.
+- A `display_name` to be displayed on the webhost.
+- A new entry in the `option_definitions` dict for your World.
+By style and convention, the internal names should be snake_case. If the option supports having multiple sub_options
+such as Choice options, these can be defined with `option_my_sub_option`, where the preceding `option_` is required and
+stripped for users, so will show as `my_sub_option` in yaml files and if `auto_display_name` is True `My Sub Option`
+on the webhost. All options support `random` as a generic option. `random` chooses from any of the available
+values for that option, and is reserved by AP. You can set this as your default value but you cannot define your own
+new `option_random`.
+
+### Option Creation
+As an example, suppose we want an option that lets the user start their game with a sword in their inventory. Let's
+create our option class (with a docstring), give it a `display_name`, and add it to a dictionary that keeps track of our
+options:
+
+```python
+class StartingSword(Toggle):
+    """Adds a sword to your starting inventory."""
+    display_name = "Start With Sword"
+
+
+options = {
+    "starting_sword": StartingSword
+}
+```
+
+This will create a `Toggle` option, internally called `starting_sword`. To then submit this to the multiworld, we add it
+to our world's `__init__.py`:
+
+```python
+from worlds.AutoWorld import World
+from .options import options
+class ExampleWorld(World):
+    option_definitions = options
+```
+
+### Option Checking
+Options are parsed by `Generate.py` before the worlds are created, and then the option classes are created shortly after
+world instantiation. These are created as attributes on the MultiWorld and can be accessed with
+`self.multiworld.my_option_name[self.player]`. This is the option class, which supports direct comparison methods to
+relevant objects (like comparing a Toggle class to a `bool`). If you need to access the option result directly, this is
+the option class's `value` attribute. For our example above we can do a simple check:
+```python
+if self.multiworld.starting_sword[self.player]:
+    do_some_things()
+```
+
+or if I need a boolean object, such as in my slot_data I can access it as:
+```python
+start_with_sword = bool(self.multiworld.starting_sword[self.player].value)
+```
+
+## Generic Option Classes
+These options are generically available to every game automatically, but can be overridden for slightly different
+behavior, if desired. See `worlds.soe.options.py` for an example.
+
+### Accessibility
+Sets rules for availability of locations for the player. Items is for all items available but not necessarily all
```

Comment:
since it is a "keyword", wrapping Items in \`\` is probably easier on the eyes.
```suggestion
Sets rules for availability of locations for the player. `Items` is for all items available but not necessarily all
```